File: leetcode/golang/240_search_a_2d_matrix_ii.go
func searchMatrix(matrix [][]int, target int) bool {
    i, j := len(matrix) - 1, 0
    for i >= 0 && j < len(matrix[0]) {
        if matrix[i][j] > target {
            i--
        } else if matrix[i][j] < target {
            j++
        } else {
            return true
        }
    }
    return false
}

File: leetcode/c++/240_search_a_2d_matrix_ii.cpp
// https://leetcode.cn/problems/search-a-2d-matrix-ii/description

#include <bits/stdc++.h>
using namespace std;

class Solution {
public:
    bool searchMatrix(vector<vector<int>>& matrix, int target) {
        int i = matrix.size() - 1, j = 0;
        while (i >= 0 && j < matrix[0].size()) {
            if (matrix[i][j] > target) i--;
            else if (matrix[i][j] < target) j++;
            else return true;
        }
        return false;
    }
};

//时间复杂度为O(m+n)，空间复杂度为O(1)